system/ee/ExpressionEngine/Controller/Design/Copy.php

```diff
@@ -0,0 +1,166 @@
+use ExpressionEngine\Model\Channel\ChannelField;
+
+/**
+* Design Controller
+*/
+class Copy extends AbstractDesignController
+{
+    public function fields(int $id)
+    {
+        $field = ee('Model')->get('ChannelField', $id)->first();
+
+        if (! $field) {
+            show_404();
+        }
+
+        // Get the template data
+        $data = [
+            'field' => $field->field_name,
+            'template_group' => '',
+            'templates' => ['all']
+        ];
+
+        // Register all template generators, and include ones disabled for template generation
+        ee('TemplateGenerator')->setCallLocation('copy')->registerAllTemplateGenerators();
+
+        $generator = ee('TemplateGenerator')->make('channel:fields');
+        $validationResult = $generator->validatePartial($data);
+        $result = $generator->generate($data, false);
+
+        // return the template_data
+        return $result['templates']['index']['template_data'];
+    }
+
+    public function channels(int $id)
+    {
+        $channel = ee('Model')->get('Channel', $id)
+        ->first();
+
+        if (! $channel) {
+            show_404();
+        }
+
+        // Get the template data
+        $data = [
+            'channel' => $channel->channel_name,
+            'template_group' => '',
+            'templates' => ['all']
+        ];
+
+        // Register all template generators, and include ones disabled for template generation
+        ee('TemplateGenerator')->setCallLocation('copy')->registerAllTemplateGenerators();
+
+        $generator = ee('TemplateGenerator')->make('channel:channels');
+        $validationResult = $generator->validatePartial($data);
+        $result = $generator->generate($data, false);
+
+        // return the template_data
+        return $result['templates']['index']['template_data'];
+    }
+
+    public function fieldGroups(int $id)
+    {
+        $field_group = ee('Model')->get('ChannelFieldGroup', $id)->first();
+
+        if (! $field_group) {
+            show_404();
+        }
+
+        // Get the template data
+        $data = [
+            'field_group' => $field_group->group_name,
+            'template_group' => '',
+            'templates' => ['all']
+        ];
+
+        // Register all template generators, and include ones disabled for template generation
+        ee('TemplateGenerator')->setCallLocation('copy')->registerAllTemplateGenerators();
+        $generator = ee('TemplateGenerator')->make('channel:fieldGroups');
+
+        $validationResult = $generator->validatePartial($data);
+        $result = $generator->generate($data, false);
+
+        // return the template_data
+        return $result['templates']['index']['template_data'];
+    }
+
+    public function fluid($fluid_id, $context, $content_id)
+    {
+        $fluidField = ee('Model')->get('ChannelField', $fluid_id)->filter('field_type', 'fluid_field')->first();
```

Comment:
Should we confirm fluid_id is an int here?

Reply:
Yeah we can use the type declaration to set it as an int.